assembl/views/api2/auth.py

```diff
@@ -490,6 +484,14 @@ def assembl_register_user(request):
     if errors:
         raise errors
 
+    # This logic needs to be above the JSONError checks to ensure that whitelisting is applied
+    # even if the discusison does not have a P_SELF_REGISTER on system.Everyone
```

Comment:
typo: discusision->discussion.
I'm trying to understand what I was doing: if the person was not allowed to self-register, I would let them register generically. I'm not 100% that was the right behaviour to start with, maybe I should just have errored.
OTH, what you're doing means that, if forbidden by a non-empty whitelist, they just fail instead of registering generically. I'm quite sure that's the right call, either way.

Reply:
Yeah, I'm being aggressive with the whitelist on register. I think it's the right call.